test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -1298,6 +1300,45 @@
             (Thread/sleep 100))
           (is (= {:received-commands 1 :executed-commands 1} (stats))))))))
 
+(deftest date-round-trip
+  (svc-utils/with-puppetdb-instance
+    (let [pdb (get-service svc-utils/*server* :PuppetDBServer)
+          dispatcher (get-service svc-utils/*server* :PuppetDBCommandDispatcher)
+          shared-globals (partial cli-svc/shared-globals pdb)
+          enqueue-command (partial enqueue-command dispatcher)
+          globals (shared-globals)
+          {{:keys [connection endpoint]} :command-mq} globals
+          deactivate-ms 14250331086887
+          ;; The problem only occurred if you passed a Date to
+          ;; enqueue, a DateTime wasn't a problem.
+          input-stamp (java.util.Date. deactivate-ms)
+          expected-stamp (DateTime. deactivate-ms DateTimeZone/UTC)]
+      (enqueue-command connection endpoint
```

Comment:
Now that enqueue is a method, I was just testing the relevant subsystem more directly, since we want it to work no matter how you reach it, but if we'd prefer to involve the http endpoint, then perhaps the test should also move to http.comand-test.